woocommerce/woocommerce-ios
# [Manual Taxes] Educational Dialog

Closes: #10515 


## Description

With this PR we show the tax educational dialog as part of the Manual Taxes project. 

## Changes
- Refactor `EditableOrderViewModel` so we can generate the `TaxEducationalDialogViewModel` with the right parameters. We pass the tax lines and the "tax based on" setting so they can be displayed accordingly.
- Create the TaxEducationalDialogView
- Create the `TaxEducationalDialogViewModel`
- Display the tax section of wp-admin when the button is tapped.
- Reload the order and the "tax based on" setting whenever the wp-admin web view is closed.
- Unit tests.


## Testing instructions


1. Go to orders
2. Tap on + to create a new order
3. Tap on the ? button on the Taxes Total row.
4. See that we show the tax educational dialog, without the tax lines or "tax based on" setting information.
5. Add products or fees so tax lines are created (make sure you have tax rates in wp-admin/admin.php?page=wc-settings&tab=tax and the products are taxable).
6. Tap again on the ? button on the Taxes Total row.
7. See that the tax lines and "tax based on" setting information is displayed.
8. Tap on the _Edit Tax Rates in Admin_ button. See that you go to the Tax settings in wp-admin. 
9. Close it. See that the order is refreshed so we can update any information related to taxes.

## Screenshots

### Educational Dialog with and without tax lines:

https://github.com/woocommerce/woocommerce-ios/assets/1864060/f927aaa3-1866-4f9f-8317-6ffc304b8d68

### The order is refreshed after closing the tax settings in wp-admin:

https://github.com/woocommerce/woocommerce-ios/assets/1864060/784c702f-332c-4489-bd57-9d96ec93cef0

---
